### PR TITLE
Bug hunt: fix 7 verified defects (generator dispatch/discovery, parallel cancellation, hook SQL)

### PR DESCRIPTION
**Pull request by behl1anmol**
## Summary

An adversarial bug hunt across the whole codebase — three parallel reviewers (source generator / runtime libraries / tests+benchmarks+CI+hooks) plus an independent full read of `HandlerDiscoveryGenerator.cs` and the runtime sources. The runtime libraries, tests, benchmarks, and CI came out **clean**; the real bugs were all in the source generator's discovery/emission edge cases, plus one hook script.

Every fix is TDD: the pinning test was written first and confirmed to fail for the documented reason before the fix. Branch ends at **125/125 tests passing** and a **warning-clean Release build** (this repo treats warnings as errors). One commit per fix.

## Fixes (severity order)

| # | Severity | Defect | How you'd hit it |
|---|----------|--------|------------------|
| **B1** | Medium | Generic/nested-generic handlers emitted unbound type parameters → **consumer build breaks** (CS0246) inside a `.g.cs` file, no diagnostic | Any generic handler class, an open generic handler (the MediatR pattern), or a handler nested in a generic type |
| **B2** | Medium | Covariant multi-response dispatch returned the **wrong handler's result** (boxed value counted as variance-assignable) | `IRequest<object> r = new MultiResponseQuery(5)` returned boxed `50` instead of `"value:5"` |
| **B3** | Low/Med | `Parallel` notifications ran every handler under an **already-cancelled** publish token; the other three strategies refuse | `cts.Cancel(); await mediator.PublishAsync(evt, cts.Token)` with ct-agnostic handlers |
| **B4** | Low/Med | A validator implementing `IValidator<T>` for several request types **silently validated only the first** | `class Shared : AbstractValidator<CreateOrder>, IValidator<UpdateOrder>` left `UpdateOrder` unvalidated |
| **B6** | Low | Closed behaviors for request types with no handler were registered in DI and counted in `BehaviorCount` (dead wiring) | Orphan `IPipelineBehavior<RetiredCommand, T>` after its handler was removed |
| **B7** | Low (breaking) | Generator honored the obsolete `[MediatorGeneration(Skip = true)]` while docs said discovery is unconditional | Contradiction either side could burn on |
| **B5** | Low | `00-save-context.csx` spliced the session id (external env var) into SQL literals + the snapshot filename | A quoted `MEDIATORLITE_SESSION_ID` broke every snapshot query, swallowed as a warning |

## Decisions baked in (confirmed with the maintainer)

- **B6** — filter dead behaviors (matches the existing validator policy); pinned counts unaffected.
- **B7** — make the code match the docs: the three `Skip` checks are removed and the attribute is now **inert**. The attribute *type* stays in Abstractions with its `[Obsolete]` marking (public-API contract). This is a **breaking** attribute-semantics change (rule 90 §3): shipped with an ADR (`.github/Memories/mediatorgeneration-skip-attribute-inert.md`), a migration-guide entry, and corrected README/quick-start/migration-from-mediatr docs that still advertised `Skip` as working. Exclusion is now structural — put the type in an assembly the generator doesn't run on.

## Not in scope

Four informational nits (immutability hardening on `ValidationException.Errors`, `Microsoft.Extensions.*` 9.0.0 on net10.0, a doc phrasing, two comment typos) were left untouched per the agreed scope.

## New diagnostic

`MEDL1004` (warning) — a handler class that is generic or nested inside a generic type is skipped instead of emitting non-compiling code. Registered in `AnalyzerReleases.Unshipped.md`.

## Learning artifacts

- Lesson: `.github/Lessons/2026-07-12-generator-discovery-guards-must-be-shared.md`
- Memory/ADR: `.github/Memories/mediatorgeneration-skip-attribute-inert.md`

## Verification

- `dotnet build MediatorLite.sln -c Release` → 0 warnings / 0 errors
- `dotnet test MediatorLite.sln` → 125/125 passing (was 116 at baseline; +9 new pinning tests)
- Each generator fix has a driver/dispatch test confirmed failing before and passing after
- Incrementality test (`UnrelatedEdit_LeavesGeneratorOutputsCached`) still passes after the B4 validator-model change
- B5 hook exercised end-to-end: valid id writes a real snapshot; `x' OR '1'='1` is refused with no file written

> Note: this environment can't execute the repo's `.claude` commit/push gates (`dotnet-script` can't run here, so the shim fails open by design). The build + test gates were run manually before every commit and the push, so their intent was honored — but on machines like this those gates are silently inactive, worth knowing.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_016jUjFYkZ9pmYQdZgU8whDD)_

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -195,15 +195,10 @@ services.AddMediatorLite();  // Without AddGeneratedHandlers(), dispatch throws
...
-
-```csharp
-[MediatorGeneration(Skip = true)]
-public class TestOnlyHandler : IRequestHandler<TestQuery, string>
-{
-    // This handler will NOT be registered by AddGeneratedHandlers()
-}
-```
+Discovery is unconditional: every concrete handler in the compilation is registered. The
+legacy `[MediatorGeneration(Skip = true)]` attribute is obsolete and has **no effect**. To
```
[the remark this answers is not in the document]
Comment (behl1anmol) [reply README.md:199]: Good catch — fixed in ec0b6a9. The packed generator README now correctly states the attribute is inert.

While fixing it I swept the rest of the repo for the same claim and found it duplicated in six more places across the `.claude`/`.cursor`/`.agents` skill-doc mirrors (`mediatorlite-abstractions`, `mediatorlite-tests`, `mediatorlite-source-generation` — each independently copied, not symlinked), all asserting the generator "still honors"/"silently honors"/"still respects" `Skip = true`. Corrected all of them in the same commit so no doc surface contradicts the B7 change.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by behl1anmol — COMMENTED** [review]
Ready to merge
